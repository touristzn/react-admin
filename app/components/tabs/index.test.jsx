import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Tabs, Tab } from './index'

let wrapper

describe('<Tabs />', () => {
  const props = {
    activeIndex: 0,
    onTabChange: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(
      <Tabs {...props}>
        <Tab>1</Tab>
        <Tab>2</Tab>
      </Tabs>
    )
  })

  it('should render two Tab component, first one should be active', () => {
    expect(wrapper.find(Tab)).toHaveLength(2)
    expect(wrapper.find('.tab-link')).toHaveLength(2)
    expect(wrapper.state('activeIndex')).toEqual(0)
    expect(wrapper.find('.tab-link').first().hasClass('active')).toEqual(true)
  })

  it('click the 2nd Tab should change the active stauts and trigger the right function', () => {
    wrapper.find('.tab-link').last().simulate('click', { preventDefault: () => {}})
    expect(wrapper.find('.tab-link').first().hasClass('active')).toEqual(false)
    expect(wrapper.find('.tab-link').last().hasClass('active')).toEqual(true)
    expect(wrapper.state('activeIndex')).toEqual(1)
    expect(props.onTabChange).toHaveBeenCalledWith(1)
  })

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
