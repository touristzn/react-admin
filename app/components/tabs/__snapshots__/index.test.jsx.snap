// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> renders correctly 1`] = `
<ul
  className="nav-tabs"
>
  <li
    className="tabs-item"
    key=".0"
  >
    <a
      className="tab-link active"
      onClick={[Function]}
    >
      <Tab>
        1
      </Tab>
    </a>
  </li>
  <li
    className="tabs-item"
    key=".1"
  >
    <a
      className="tab-link"
      onClick={[Function]}
    >
      <Tab>
        2
      </Tab>
    </a>
  </li>
</ul>
`;
